velox/expression/CastExpr-inl.h

```diff
@@ -249,11 +250,14 @@ VectorPtr CastExpr::applyDecimalToIntegralCast(
   applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
     auto value = simpleInput->valueAt(row);
     auto integralPart = value / scaleFactor;
-    auto fractionPart = value % scaleFactor;
-    auto sign = value >= 0 ? 1 : -1;
-    bool needsRoundUp =
-        (scaleFactor != 1) && (sign * fractionPart >= (scaleFactor >> 1));
-    integralPart += needsRoundUp ? sign : 0;
+    if (!queryConfig.isCastToIntByTruncate()) {
```

Comment:
This function is called for each row in a hot loop. Let's introduce a temp variable and initialize it outside of the loop (similar to scaleFactor).

Reply:
Updated.